### PR TITLE
Bug 1881571 - Add tooling-lint component to app projects

### PR DESCRIPTION
**Pull request by jonalmeida**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1881571

### CODE REVIEW

File: android-components/components/tooling/lint/src/main/java/mozilla/components/tooling/lint/ImageViewAndroidTintXmlDetector.kt [hunk 1]
```diff
@@ -36,7 +35,6 @@ class ImageViewAndroidTintXmlDetector : ResourceXmlDetector() {
         const val ERROR_MESSAGE =
             "Using android:tint to tint ImageView instead of app:tint with AppCompatImageView"
 
-        @VisibleForTesting
```
Comment (jonalmeida) [inline android-components/components/tooling/lint/src/main/java/mozilla/components/tooling/lint/ImageViewAndroidTintXmlDetector.kt:39]: When doing a checksum diff between Fenix and the AC component, this was a difference and didn't need to be there. Removing it to ensure that when BZ-1872229 is worked on, its a tiny bit easier for the next person to de-duplicate and save them the trouble.

**Comment by jonalmeida** [comment]
cc: @jackyzy823 - this should ensure the new linter is used in the app projects as well.

**Review by Amejia481 — APPROVED** [review]

**Comment by jonalmeida** [comment]
> Will the lint-baseline.xml stops working if line number changes? 

This baseline file is for Android Lint, and we use a baseline in AC and Fenix today without issues. The concerns you are referring to were with ktlint baseline which isn't smart enough for some reason to not cause a problem.

I've also generated a baseline for Focus instead of fixing any of these issues because we don't actively maintain this project unless needed.

@rocketsroger do you think we should fix these issues in a fast follow-up regardless?

cc: @mcarare who might have experience here as well.

**Comment by rocketsroger** [comment]
> @rocketsroger do you think we should fix these issues in a fast follow-up regardless?
> 
> cc: @mcarare who might have experience here as well.'

I think we should add the fix add the same time with this change so we don't need baseline.

**Comment by mcarare** [comment]
Although the line numbers are mentioned in the baseline, for lint it should not matter if they are changed in the future.
I do think we should fix the issues, at least the errors added to the baseline.


I do see a lot of issues related to strings, where we have a string in one language but no value in the default string.xml. Is this something L10n team should look into?

**Comment by rocketsroger** [comment]
> Although the line numbers are mentioned in the baseline, for lint it should not matter if they are changed in the future. I do think we should fix the issues, at least the errors added to the baseline.

I'm OK with merging this then.  Thanks

**Pull request merged**